File: gradle-plugin/src/main/java/juuxel/ripple/gradle/RipplePlugin.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package juuxel.ripple.gradle;

import juuxel.ripple.gradle.impl.RippleExtensionImpl;
import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class RipplePlugin implements Plugin<Project> {
    @Override
    public void apply(Project target) {
        RippleExtensionImpl extension = (RippleExtensionImpl) target.getExtensions().create(RippleExtension.class, "ripple", RippleExtensionImpl.class);
        extension.detachedConfigurationGetter = target.getConfigurations()::detachedConfiguration;
        extension.cache = target.getGradle().getGradleUserHomeDir().toPath().resolve("caches").resolve("ripple-cache");
        extension.fileCollectionCreator = target::files;
        extension.fileResolver = target::file;
        extension.dependencyCreator = target.getDependencies()::create;
        extension.refreshDeps = target.getGradle().getStartParameter().isRefreshDependencies();
    }
}

File: gradle-plugin/src/main/kotlin/juuxel/ripple/gradle/RipplePlugin.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package juuxel.ripple.gradle

import org.gradle.api.Plugin
import org.gradle.api.Project

class RipplePlugin : Plugin<Project> {
    override fun apply(target: Project) {
        val extension = target.extensions.create("ripple", RippleExtension::class.java)
        extension.detachedConfigGetter = {
            target.configurations.detachedConfiguration(it)
        }
        extension.cache = target.projectDir.toPath().resolve(".gradle").resolve("ripple-cache")
        extension.fileCollectionGetter = { target.files(it) }
        extension.fileResolver = target::file
        extension.dependencyCreator = { target.dependencies.create(it) }
    }
}
